src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -20,22 +23,147 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Lotus/BgLotus_Init.s")
+s32 D_80AD6D10[] = { 0x48580064 };
+
+void BgLotus_Init(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+
+    s32 pad;
+    s32 sp2C;
+
+    Actor_ProcessInitChain(&this->dyna.actor, (InitChainEntry*)D_80AD6D10);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gLilyPadCol);
+    this->dyna.actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->dyna.actor.floorPoly, &sp2C,
+                                                               &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->unk168 = 0x60;
+    this->dyna.actor.world.rot.y = ((s32)Rand_Next() >> 0x10);
+    this->actionFunc = func_80AD68DC;
+}
+
+void BgLotus_Destroy(Actor* thisx, PlayState* play) {
+    BgLotus* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80AD6830(Actor* thisx) {
```

Comment:
```suggestion
void func_80AD6830(BgLotus* this) {
```
Since this is an overlay function, it makes more sense for it to take the `BgLotus` struct rather than the base actor.